funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -363,15 +364,31 @@ def process_pending_tasks():
 
                     try:
                         incoming_task = self.pending_task_queue.get(timeout=1)
-                        task = try_convert_from_messagepack(incoming_task)
-                        executor.submit_raw(task)
-                        num_tasks_forwarded += 1  # Safe given GIL
+                        task_msg = unpack(incoming_task)
+                        if not isinstance(task_msg, Task):
+                            raise InvalidMessageError()
 
                     except queue.Empty:
                         continue
 
                     except Exception:
-                        log.exception("Unhandled issue while waiting for pending tasks")
+                        log.exception("Unhandled error processing incoming task")
+                        continue
+
+                    try:
+                        task = convert_to_internaltask(task_msg)
+                        executor.submit_raw(task)
+                        num_tasks_forwarded += 1  # Safe given GIL
+
+                    except Exception as exc:
+                        log.exception(f"Failed to process {task_msg.task_id}")
+                        result = {
```

Comment:
Reporting a failure here is a good improvement. This would allow us to immediately fail out tasks sent to an executor in a broken state.